File: Project/JWT/ManagementParking/BussinessLayer/Services/Interface/IAuthService.cs
using BussinessLayer.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BussinessLayer.Services.Interface
{
    public interface IAuthService
    {
        Task<AuthResponse> RegisterAsync(RegisterDto dto);
        Task<AuthResponse> LoginAsync(LoginDto dto);
        Task<bool> ChangePasswordAsync(ChangePasswordDto dto);
        Task<bool> ForgotPasswordAsync(string email);

        Task<AuthResponse> RefreshTokenAsync(string refreshToken);
    }

}

File: VBNet/MyLibrary.BLL/Services/Interface/IAuthService.vb
Public Interface IAuthService
    Function LoginAsync(email As String, password As String) As Task(Of LoginResponseDto)

    Function RegisterAsync(dto As RegisterRequestDto) As Task

    Function EmailExistsAsync(email As String) As Task(Of Boolean)

    Function VerifyAccountAsync(email As String, code As String) As Task

    Function ForgotPasswordAsync(email As String) As Task

    Function CompletePasswordResetAsync(email As String, otp As String, newPassword As String) As Task
End Interface